langserver/symbol.go

```diff
@@ -220,10 +212,21 @@ func toSym(name, container string, kind lsp.SymbolKind, fs *token.FileSet, pos t
 	}
 }
 
+// handleTextDocumentSymbol handles `textDocument/documentSymbol` requests for
+// the Go language server.
+func (h *LangHandler) handleTextDocumentSymbol(ctx context.Context, conn JSONRPC2Conn, req *jsonrpc2.Request, params lsp.DocumentSymbolParams) ([]lsp.SymbolInformation, error) {
+	f := strings.TrimPrefix(params.TextDocument.URI, "file://")
+	return h.handleSymbol(ctx, conn, req, query{file: f}, 0)
```

Comment:
This could probably be made more efficient by only collecting the symbols from the file in question, instead of collecting all symbols and then filtering by file (`handleSymbol` iterates over each package and calls `getPackageSyms` for every package regardless of whether it contains the desired file).